pinot-core/src/main/java/org/apache/pinot/core/data/aggregator/SumPrecisionValueAggregator.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.aggregator;
+
+import java.math.BigDecimal;
+import org.apache.pinot.common.function.AggregationFunctionType;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.apache.pinot.spi.utils.BigDecimalUtils;
+
+
+public class SumPrecisionValueAggregator implements ValueAggregator<Object, BigDecimal> {
+  public static final DataType AGGREGATED_VALUE_TYPE = DataType.BYTES;
+
+  private int _maxByteSize;
```

Comment:
Should this be initialized to 0 ?

Reply:
It doesn't really matter as we won't generate star-tree if there is no record. Also, without explicitly putting a value, member variable will be initialized to 0